backend/function-apps/api/case-notes/case.notes.function.test.ts

```diff
@@ -0,0 +1,198 @@
+import handler from './case.notes.function';
+import { CaseNotesController } from '../../../lib/controllers/case-notes/case.notes.controller';
+import ContextCreator from '../../azure/application-context-creator';
+import { MockData } from '../../../../common/src/cams/test-utilities/mock-data';
+import { CamsHttpRequest } from '../../../lib/adapters/types/http';
+import { InvocationContext } from '@azure/functions';
+import {
+  buildTestResponseError,
+  buildTestResponseSuccess,
+  createMockAzureFunctionRequest,
+} from '../../azure/testing-helpers';
+import { UnknownError } from '../../../lib/common-errors/unknown-error';
+import { NORMAL_CASE_ID } from '../../../lib/testing/testing-constants';
+import { CaseNote } from '../../../../common/src/cams/cases';
+import * as featureFlags from '../../../lib/adapters/utils/feature-flag';
+
+const defaultRequestProps: Partial<CamsHttpRequest> = {
+  method: 'POST',
+  body: {
+    caseId: '081-67-89123',
+    note: 'Sample note text',
+  },
+};
+
+describe('Case Notes Function Tests', () => {
+  let context;
+
+  beforeEach(() => {
+    jest
+      .spyOn(ContextCreator, 'getApplicationContextSession')
+      .mockResolvedValue(MockData.getManhattanAssignmentManagerSession());
+    context = new InvocationContext({
+      logHandler: () => {},
+      invocationId: 'id',
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('should handle successful response', async () => {
+    const req = createMockAzureFunctionRequest();
+    const { azureHttpResponse, camsHttpResponse } = buildTestResponseSuccess({
+      meta: {
+        self: req.url,
+      },
+      data: undefined,
+    });
+    jest.spyOn(CaseNotesController.prototype, 'handleRequest').mockResolvedValue(camsHttpResponse);
+
+    const response = await handler(req, context);
+    expect(response).toEqual(azureHttpResponse);
+  });
+
+  test('should handle error', async () => {
+    const req = createMockAzureFunctionRequest();
+    const error = new Error('Some unknown error');
+    const { azureHttpResponse } = buildTestResponseError(error);
+    jest.spyOn(CaseNotesController.prototype, 'handleRequest').mockRejectedValue(error);
+    const response = await handler(req, context);
+    expect(response).toEqual(azureHttpResponse);
+  });
+
+  // TODO: did we need this stuff for a controller test?
```

Comment:
If we aren't going to use what is below this somewhere else, let's pull it out since it's unneeded for testing the Function.